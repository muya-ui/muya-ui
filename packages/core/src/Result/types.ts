import { HTMLAttributes, ReactNode } from 'react';

import { IImgProps } from '../Img';
import { ICustomStyleBaseProps } from '../types';

/**
 * @styles content 内容
 * @styles icon icon
 * @styles title 标题
 * @styles subTitle 子标题
 * @styles reason 原因
 * @styles extra 额外的
 */
export type IResultStyleKeys = 'content' | 'icon' | 'title' | 'subTitle' | 'reason' | 'extra';

export type IResultIconType = 'success' | 'error' | 'warning' | 'info';

export type IResultStatusIcon = 'forbidden' | 'empty' | 'emptySmall';

export interface IResultPureProps extends ICustomStyleBaseProps<IResultStyleKeys> {
  /**
   * 标题
   */
  title?: string;
  /**
   * 副标题
   */
  subTitle?: string;
  /**
   * 错误原因
   */
  reason?: ReactNode;
  /**
   * 类型
   * success-成功 error-失败 warning-警告 info-提示
   * 403-无权限 empty-空数据 emptySmall-空数据小Icon
   * @default 'info'
   */
  type?: 'success' | 'error' | 'warning' | 'info' | 'forbidden' | 'empty' | 'emptySmall';
  /**
   * 布局是否垂直
   * @default true
   */
  vertical?: boolean;
  /**
   * 自定义 icon
   */
  icon?: ReactNode;
  /**
   * icon 大小
   */
  iconSize?: number;
  /**
   * 操作区
   */
  extra?: ReactNode;
  /**
   * icon 传入字符串时可以自定义图片属性
   */
  imgProps?: IImgProps;
}

export type IResultProps = IResultPureProps & HTMLAttributes<HTMLDivElement>;
