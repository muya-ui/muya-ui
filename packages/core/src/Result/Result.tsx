import React, { forwardRef } from 'react';

import useLocale from '../Locale/useLocale';
import {
  ClearIcon,
  InformIcon,
  IResultStatusIconType,
  ReminderIcon,
  SuccessIcon,
} from '@muya-ui/theme-light';

import { ImgImg } from '../Img';
import Typography from '../Typography';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import {
  StyleButton,
  StyledIcon,
  StyledResult,
  StyleReason,
  StyleSubTitle,
  StyleTitle,
} from './styled';
import { IResultIconType, IResultProps } from './types';

export const Icons = ['success', 'error', 'info', 'warning'];

export const ExceptionStatus = ['forbidden', 'empty', 'emptySmall'];

const Result = forwardRef<HTMLDivElement, IResultProps>((props, ref) => {
  const theme = useTheme();
  const {
    components: { Result: token },
  } = theme;
  const {
    title,
    subTitle,
    icon,
    type = 'info',
    iconSize = token.defaultIconSize,
    vertical = true,
    extra,
    reason,
    styles,
    imgProps,
    ...otherProps
  } = props;
  const local = useLocale();
  const innerStyles = useStyles(
    'result',
    {
      content: '',
      icon: '',
      title: '',
      subTitle: '',
      reason: '',
      extra: '',
    },
    styles,
  );
  let iconNode;
  const isStatusIcon = ExceptionStatus.includes(type);
  const typeIcon = {
    success: SuccessIcon,
    info: InformIcon,
    warning: ReminderIcon,
    error: ClearIcon,
    forbidden:
      '//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737658/62EC08BDA2CDEA69697790DCB6240015.svg',
    empty:
      '//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737659/28AD5A0CE2237E2987B3D751C4C5BA83.svg',
    emptySmall:
      '//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737281/D8307EB1E1C8C21BCE88D2D55B980DE5.svg',
    ...theme.components.Result.typeIcon,
  };
  if (isStatusIcon) {
    const icon = typeIcon[type] as string;
    const { width, height } = token.typeIconSize[type as IResultStatusIconType];
    iconNode = (
      <StyledIcon
        {...innerStyles.icon}
        theme={theme}
        $vertical={vertical}
        $type={type}
        $iconSize={height}
      >
        <img src={icon} width={width} alt="" />
      </StyledIcon>
    );
  } else {
    if (React.isValidElement(icon)) {
      iconNode = icon;
    } else {
      const Icon = typeIcon[type];
      const iconBgColor = token.typeBg[type as IResultIconType];
      iconNode = (
        <StyledIcon
          {...innerStyles.icon}
          theme={theme}
          $vertical={vertical}
          $type={type}
          $iconSize={iconSize}
        >
          {typeof icon === 'string' ? (
            <ImgImg src={icon} suffixWidth={iconSize} {...imgProps} />
          ) : (
            <Icon bgColor={iconBgColor} fontSize={iconSize} />
          )}
        </StyledIcon>
      );
    }
  }
  let renderTitle = title;
  let renderSubTitle = subTitle;
  if (type === 'forbidden') {
    renderTitle = renderTitle || '403';
    renderSubTitle = renderSubTitle || local['Result.forbiddenSubTitle'];
  }
  if (type.includes('empty')) {
    renderSubTitle = renderSubTitle || local['Result.emptyText'];
  }

  return (
    <StyledResult ref={ref} theme={theme} $vertical={vertical} {...otherProps}>
      {iconNode}
      <div {...innerStyles.content}>
        {renderTitle ? (
          <StyleTitle
            {...innerStyles.title}
            theme={theme}
            $vertical={vertical}
            $iconSize={iconSize}
          >
            <Typography.Title strong level={token.defaultTitleLevel}>
              {renderTitle}
            </Typography.Title>
          </StyleTitle>
        ) : null}
        {renderSubTitle ? (
          <StyleSubTitle {...innerStyles.subTitle} theme={theme}>
            <Typography.Paragraph fontSize={token.defaultSubTitleFontLevel} color="assistant">
              {renderSubTitle}
            </Typography.Paragraph>
          </StyleSubTitle>
        ) : null}
        {reason ? (
          <StyleReason {...innerStyles.reason} theme={theme}>
            {reason}
          </StyleReason>
        ) : null}
        {extra ? (
          <StyleButton {...innerStyles.extra} theme={theme}>
            {extra}
          </StyleButton>
        ) : null}
      </div>
    </StyledResult>
  );
});

export default Result;
