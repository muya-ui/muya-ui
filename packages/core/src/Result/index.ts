export { default } from './Result';
export { default as Result } from './Result';
export * from './types';
