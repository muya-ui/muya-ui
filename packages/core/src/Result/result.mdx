---
name: Result 结果页
menu: 组件
route: /components/result
autoDemos: true
parent: 反馈
---

# Result 结果页

用于反馈一些列任务操作、无数据状态的结果。

## 使用场景

- 当有重要操作需告知用户处理结果，且反馈内容较为复杂时使用
- 当目前没有数据时，用于用户提示
- 需要对结果有后续的引导

{Demos}

### API

#### IResultPureProps

<Props propsName="IResultProps" />

### Styles

#### IResultStyleKeys

<Styles typeName="IResultStyleKeys" componentName="result" />
