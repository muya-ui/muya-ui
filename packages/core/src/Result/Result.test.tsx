import React from 'react';
import renderer from 'react-test-renderer';

import { IResultIconType, TimeIcon } from '@muya-ui/theme-light';
import { Button, Link } from '@muya-ui/core';

import Result from './Result';
import { IResultProps } from './types';

let id = 0;

const renderResult = ({
  title,
  subTitle,
  type,
  extra,
  reason,
  vertical,
  icon,
  iconSize,
}: IResultProps) => (
  <Result
    key={id++}
    title={title}
    subTitle={subTitle}
    type={type}
    extra={extra}
    reason={reason}
    vertical={vertical}
    icon={icon}
    iconSize={iconSize}
  />
);

const Buttons = (
  <>
    <Button plain={false}>次按钮</Button>
    <Button type="primary">主按钮</Button>
  </>
);

const Reason = (
  <>
    错误原因
    <Link
      href="/"
      style={{
        margin: '0 5px',
        textDecoration: 'none',
        color: '#1a7af8',
      }}
    >
      文字链
    </Link>
  </>
);

const customIcon = <TimeIcon fontSize={50} />;

describe('Result Component', () => {
  it('should Result Component mount success', () => {
    const wrapper = renderer
      .create(
        renderResult({
          title: '提示',
          subTitle: '提示文案',
          type: 'success',
          extra: Buttons,
          reason: Reason,
          iconSize: 100,
        }),
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render different type', () => {
    const wrapper = renderer
      .create(
        <>
          {['success', 'error', 'warning', 'info', 'forbidden', 'empty', 'emptySmall'].map(item =>
            renderResult({
              type: item as IResultIconType | IResultIconType,
            }),
          )}
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render default text', () => {
    const wrapper = renderer
      .create(
        <>
          {['forbidden', 'empty', 'emptySmall'].map(item =>
            renderResult({
              type: item as IResultIconType | IResultIconType,
            }),
          )}
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render different layout', () => {
    const wrapper = renderer
      .create(
        <>
          {renderResult({
            type: 'success',
            vertical: false,
            title: '提示',
            subTitle: '提示文案',
          })}
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render custom icon', () => {
    const wrapper = renderer
      .create(
        <>
          {renderResult({
            icon: customIcon,
            iconSize: 100,
          })}
          {renderResult({
            icon:
              '//qhstaticssl.kujiale.com/newt/23/image/png/1565749312620/129F060E504FDCDAEE29088AB181D326.png',
            iconSize: 100,
          })}
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
