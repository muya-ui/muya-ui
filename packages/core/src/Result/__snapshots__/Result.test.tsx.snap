// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component should Result Component mount success 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 100;
  fill: currentColor;
}

.c14 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c16 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 32px;
  padding: 0 14px 0 14px;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: rgba(102,102,102,1);
  cursor: pointer;
  border-color: rgba(128,128,128,1);
  border-width: 0;
}

.c16:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(128,128,128,1);
  border-color: rgba(128,128,128,1);
}

.c16:active {
  color: rgba(255,255,255,1);
  background-color: rgba(89,89,89,1);
  border-color: rgba(128,128,128,1);
}

.c16 + .c15 {
  margin-left: 8px;
}

.c17 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 32px;
  padding: 0 14px 0 14px;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: rgba(26,122,248,1);
  cursor: pointer;
  border-color: rgba(51,139,255,1);
  border-width: 0;
}

.c17:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(51,139,255,1);
  border-color: rgba(51,139,255,1);
}

.c17:active {
  color: rgba(255,255,255,1);
  background-color: rgba(12,108,235,1);
  border-color: rgba(51,139,255,1);
}

.c17 + .c15 {
  margin-left: 8px;
}

.c11 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(26,122,248,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c11:hover {
  color: rgba(51,139,255,1);
}

.c11:active {
  color: rgba(12,108,235,1);
}

.c11 + .c10 {
  margin-left: 8px;
}

.c7 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 0;
}

.c5 + .c4 {
  margin-top: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(0,217,72,1);
  height: 100px;
}

.c3 {
  margin-top: 16px;
}

.c3 .c4 {
  margin-bottom: 0;
}

.c6 {
  margin-top: 8px;
}

.c8 {
  margin-top: 28px;
}

.c13 {
  margin-top: 28px;
}

.c12.c12 {
  position: static;
}

<div
  className="c0"
>
  <div
    className="c1 muya-result-icon"
  >
    <svg
      aria-hidden="false"
      className="c2"
      focusable="false"
      fontSize={100}
      viewBox="0 0 1024 1024"
    >
      <rect
        fill="rgba(1,217,73,0.08)"
        height={716.8}
        width={716.8}
        x={153.60000000000002}
        y={153.60000000000002}
      />
      <path
        d="M436.576 574.933l-105.6-105.6a21.333 21.333 0 00-30.165 0l-60.342 60.342a21.333 21.333 0 000 30.165l181.014 181.013c4.17 4.171 9.632 6.262 15.093 6.262s10.923-2.091 15.083-6.251L783.53 408.992a21.333 21.333 0 000-30.165l-60.342-60.342a21.333 21.333 0 00-30.165 0L436.576 574.923zM512 1024C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512z"
      />
    </svg>
  </div>
  <div
    className="muya-result-content"
  >
    <div
      className="c3 muya-result-title"
    >
      <h3
        className="c4 c5"
      >
        提示
      </h3>
    </div>
    <div
      className="c6 muya-result-sub-title"
    >
      <p
        className="c7"
        color="assistant"
        fontSize="s1"
      >
        提示文案
      </p>
    </div>
    <div
      className="c8 muya-result-reason"
    >
      错误原因
      <a
        aria-disabled={false}
        className="c9 c10 c11 c12"
        href="/"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "color": "#1a7af8",
            "margin": "0 5px",
            "textDecoration": "none",
          }
        }
      >
        文字链
      </a>
    </div>
    <div
      className="c13 muya-result-extra"
    >
      <button
        className="c14 c15 c16"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        次按钮
      </button>
      <button
        className="c14 c15 c17"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        主按钮
      </button>
    </div>
  </div>
</div>
`;

exports[`Result Component should render custom icon 1`] = `
Array [
  .c1 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 50;
  fill: currentColor;
}

.c0 {
  text-align: center;
}

<div
    className="c0"
  >
    <svg
      aria-hidden="false"
      className="c1"
      focusable="false"
      fontSize={50}
      viewBox="0 0 1024 1024"
    >
      <path
        d="M576 448V234.667a21.333 21.333 0 00-21.333-21.334h-85.334A21.333 21.333 0 00448 234.667v320A21.333 21.333 0 00469.333 576h320a21.333 21.333 0 0021.334-21.333v-85.334A21.333 21.333 0 00789.333 448H576zm-64 576C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512z"
      />
    </svg>
    <div
      className="muya-result-content"
    />
  </div>,
  .c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(26,122,248,1);
  height: 100px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        className="muya-img-loading"
        src=""
      />
    </div>
    <div
      className="muya-result-content"
    />
  </div>,
]
`;

exports[`Result Component should render default text 1`] = `
Array [
  .c6 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 0;
}

.c4 + .c3 {
  margin-top: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 180px;
}

.c2 {
  margin-top: 16px;
}

.c2 .c3 {
  margin-bottom: 0;
}

.c5 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737658/62EC08BDA2CDEA69697790DCB6240015.svg"
        width={180}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-title"
      >
        <h3
          className="c3 c4"
        >
          403
        </h3>
      </div>
      <div
        className="c5 muya-result-sub-title"
      >
        <p
          className="c6"
          color="assistant"
          fontSize="s1"
        >
          您暂无权限查看
        </p>
      </div>
    </div>
  </div>,
  .c3 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 180px;
}

.c2 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737659/28AD5A0CE2237E2987B3D751C4C5BA83.svg"
        width={180}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-sub-title"
      >
        <p
          className="c3"
          color="assistant"
          fontSize="s1"
        >
          暂无数据
        </p>
      </div>
    </div>
  </div>,
  .c3 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 48px;
}

.c2 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737281/D8307EB1E1C8C21BCE88D2D55B980DE5.svg"
        width={48}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-sub-title"
      >
        <p
          className="c3"
          color="assistant"
          fontSize="s1"
        >
          暂无数据
        </p>
      </div>
    </div>
  </div>,
]
`;

exports[`Result Component should render different layout 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 32;
  fill: currentColor;
}

.c7 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 0;
}

.c5 + .c4 {
  margin-top: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-right: 16px;
  color: rgba(0,217,72,1);
  height: 32px;
}

.c8 .c4 {
  margin-bottom: 0;
}

.c3 {
  margin-top: 4px;
}

.c3 .c4 {
  margin-bottom: 0;
}

.c6 {
  margin-top: 8px;
}

<div
  className="c0"
>
  <div
    className="c1 muya-result-icon"
  >
    <svg
      aria-hidden="false"
      className="c2"
      focusable="false"
      fontSize={32}
      viewBox="0 0 1024 1024"
    >
      <rect
        fill="rgba(1,217,73,0.08)"
        height={716.8}
        width={716.8}
        x={153.60000000000002}
        y={153.60000000000002}
      />
      <path
        d="M436.576 574.933l-105.6-105.6a21.333 21.333 0 00-30.165 0l-60.342 60.342a21.333 21.333 0 000 30.165l181.014 181.013c4.17 4.171 9.632 6.262 15.093 6.262s10.923-2.091 15.083-6.251L783.53 408.992a21.333 21.333 0 000-30.165l-60.342-60.342a21.333 21.333 0 00-30.165 0L436.576 574.923zM512 1024C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512z"
      />
    </svg>
  </div>
  <div
    className="muya-result-content"
  >
    <div
      className="c3 muya-result-title"
    >
      <h3
        className="c4 c5"
      >
        提示
      </h3>
    </div>
    <div
      className="c6 muya-result-sub-title"
    >
      <p
        className="c7"
        color="assistant"
        fontSize="s1"
      >
        提示文案
      </p>
    </div>
  </div>
</div>
`;

exports[`Result Component should render different type 1`] = `
Array [
  .c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 32;
  fill: currentColor;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(0,217,72,1);
  height: 32px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <svg
        aria-hidden="false"
        className="c2"
        focusable="false"
        fontSize={32}
        viewBox="0 0 1024 1024"
      >
        <rect
          fill="rgba(1,217,73,0.08)"
          height={716.8}
          width={716.8}
          x={153.60000000000002}
          y={153.60000000000002}
        />
        <path
          d="M436.576 574.933l-105.6-105.6a21.333 21.333 0 00-30.165 0l-60.342 60.342a21.333 21.333 0 000 30.165l181.014 181.013c4.17 4.171 9.632 6.262 15.093 6.262s10.923-2.091 15.083-6.251L783.53 408.992a21.333 21.333 0 000-30.165l-60.342-60.342a21.333 21.333 0 00-30.165 0L436.576 574.923zM512 1024C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512z"
        />
      </svg>
    </div>
    <div
      className="muya-result-content"
    />
  </div>,
  .c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 32;
  fill: currentColor;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(255,43,0,1);
  height: 32px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <svg
        aria-hidden="false"
        className="c2"
        focusable="false"
        fontSize={32}
        viewBox="0 0 1024 1024"
      >
        <rect
          fill="rgba(255,43,0,0.08)"
          height={716.8}
          width={716.8}
          x={153.60000000000002}
          y={153.60000000000002}
        />
        <path
          d="M602.507 512L730.73 383.787a21.333 21.333 0 000-30.187l-60.342-60.33a21.333 21.333 0 00-30.165 0L512 421.482 383.787 293.269a21.333 21.333 0 00-30.187 0l-60.33 60.342a21.333 21.333 0 000 30.165L421.482 512 293.269 640.213a21.333 21.333 0 000 30.187l60.342 60.33a21.333 21.333 0 0030.165 0L512 602.518l128.213 128.214a21.333 21.333 0 0030.187 0l60.33-60.342a21.333 21.333 0 000-30.165L602.518 512zM512 1024C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512z"
        />
      </svg>
    </div>
    <div
      className="muya-result-content"
    />
  </div>,
  .c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 32;
  fill: currentColor;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(255,171,0,1);
  height: 32px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <svg
        aria-hidden="false"
        className="c2"
        focusable="false"
        fontSize={32}
        viewBox="0 0 1024 1024"
      >
        <rect
          fill="rgba(255,171,0,0.08)"
          height={716.8}
          width={716.8}
          x={153.60000000000002}
          y={153.60000000000002}
        />
        <path
          d="M512 1024C229.227 1024 0 794.773 0 512S229.227 0 512 0s512 229.227 512 512-229.227 512-512 512zm-42.667-426.667h85.334A21.333 21.333 0 00576 576V234.667a21.333 21.333 0 00-21.333-21.334h-85.334A21.333 21.333 0 00448 234.667V576a21.333 21.333 0 0021.333 21.333zm0 213.334h85.334A21.333 21.333 0 00576 789.333V704a21.333 21.333 0 00-21.333-21.333h-85.334A21.333 21.333 0 00448 704v85.333a21.333 21.333 0 0021.333 21.334z"
        />
      </svg>
    </div>
    <div
      className="muya-result-content"
    />
  </div>,
  .c2 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  font-size: 32;
  fill: currentColor;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  color: rgba(26,122,248,1);
  height: 32px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <svg
        aria-hidden="false"
        className="c2"
        focusable="false"
        fontSize={32}
        viewBox="0 0 1024 1024"
      >
        <rect
          fill="rgba(26,122,248,0.08)"
          height={716.8}
          width={716.8}
          x={153.60000000000002}
          y={153.60000000000002}
        />
        <path
          d="M853.333 704h64a21.333 21.333 0 0121.334 21.333V832a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 832V725.333A21.333 21.333 0 01106.667 704h64V341.333C170.667 152.821 323.488 0 512 0c188.512 0 341.333 152.821 341.333 341.333V704zm-468.81 213.333h254.944C633.6 977.131 578.773 1024 512 1024s-121.6-46.87-127.477-106.667z"
        />
      </svg>
    </div>
    <div
      className="muya-result-content"
    />
  </div>,
  .c6 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 18px;
  line-height: 24px;
  margin-bottom: 0;
}

.c4 + .c3 {
  margin-top: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 180px;
}

.c2 {
  margin-top: 16px;
}

.c2 .c3 {
  margin-bottom: 0;
}

.c5 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737658/62EC08BDA2CDEA69697790DCB6240015.svg"
        width={180}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-title"
      >
        <h3
          className="c3 c4"
        >
          403
        </h3>
      </div>
      <div
        className="c5 muya-result-sub-title"
      >
        <p
          className="c6"
          color="assistant"
          fontSize="s1"
        >
          您暂无权限查看
        </p>
      </div>
    </div>
  </div>,
  .c3 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 180px;
}

.c2 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737659/28AD5A0CE2237E2987B3D751C4C5BA83.svg"
        width={180}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-sub-title"
      >
        <p
          className="c3"
          color="assistant"
          fontSize="s1"
        >
          暂无数据
        </p>
      </div>
    </div>
  </div>,
  .c3 {
  padding: 0;
  color: rgba(153,153,153,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c0 {
  text-align: center;
}

.c1 {
  margin-right: 0px;
  height: 48px;
}

.c2 {
  margin-top: 8px;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-result-icon"
    >
      <img
        alt=""
        src="//qhstaticssl.kujiale.com/newt/100759/image/svgxml/1573020737281/D8307EB1E1C8C21BCE88D2D55B980DE5.svg"
        width={48}
      />
    </div>
    <div
      className="muya-result-content"
    >
      <div
        className="c2 muya-result-sub-title"
      >
        <p
          className="c3"
          color="assistant"
          fontSize="s1"
        >
          暂无数据
        </p>
      </div>
    </div>
  </div>,
]
`;
